importer/src/module_loader.rs

```diff
@@ -1,13 +1,53 @@
-use std::path::PathBuf;
+use std::{env, path::PathBuf};
 
 use ast::parsed::{
-    asm::{ASMProgram, Module},
+    asm::{
+        ASMModule, ASMProgram, Import, Module, ModuleStatement, Part, SymbolDefinition, SymbolPath,
+        SymbolValue,
+    },
     folder::Folder,
 };
 use number::FieldElement;
+use parser::parse_asm;
 
 static ASM_EXTENSION: &str = "asm";
 static FOLDER_MODULE_NAME: &str = "mod";
+static POWDR_STD_ENV: &str = "POWDR_STD";
+static MOD_FILE: &str = "mod.asm";
+static STD_NAME: &str = "std";
+
+/// Loads the standard library module from the location specified in the <POWDR_STD_ENV> environment variable
+/// (or, if unset, <project_root>/std).
+///
+/// # Panics
+/// If there is an error loading the standard library
+fn load_std<T: FieldElement>() -> ASMModule<T> {
+    let default_std_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+        .parent()
+        .unwrap()
+        .join("std");
+    let std_path = env::var(POWDR_STD_ENV)
+        .map(PathBuf::from)
+        .unwrap_or(default_std_path);
+    let std_path = std_path.join(MOD_FILE);
+
+    match std::fs::read_to_string(&std_path) {
```

Comment:
Could we somehow include it is a resource, i.e. directly in the binary (with the option to use a different one at a specified path)?

Reply:
Yes, I think one way would be:
- In `build.rs`, load the std, resolve all external modules and write a self-contained file into `OUT_DIR`
- Include the generated file using `include_str!()`

Sounds like a good follow-up PR to me, what do you think?